### PR TITLE
Fix Files Impossible to download a video image on PRE1

### PR DESCRIPTION
**Pull request by liviu-preda-tsystems**
Added additional rules for CSP content security policy e.g. media_source 'self' , default-src 'self'
https://jira.telekom.de/browse/NMC-2675

### CODE REVIEW

File: lib/Listener/BeforeTemplateRenderedListener.php [hunk 1]
```diff
@@ -102,5 +107,9 @@ public function handle(Event $event): void {
 		\OCP\Util::addScript("nmctheme", "nmctheme-conflictdialog");
 		\OCP\Util::addScript("nmctheme", "nmctheme-tooltip");
 		\OCP\Util::addScript("nmctheme", "nmctheme-shareicons");
+
+		$csp = new \OCP\AppFramework\Http\ContentSecurityPolicy();
+		$csp->addAllowedFrameDomain('\'self\'');
+		$response->setContentSecurityPolicy($csp);
```
Comment (tsdicloud) [inline lib/Listener/BeforeTemplateRenderedListener.php]: This time, an additional CSP rule for frames is set. I could imagine that this is the only rule that actually has effect?
But why for frames?

File: lib/Listener/BeforeTemplateRenderedListener.php [hunk 2]
```diff
@@ -72,14 +73,18 @@ public function handle(Event $event): void {
...
+			$response->setContentSecurityPolicy($csp);
 		}
 
 		if (($response->getStatus() >= 400) && ($response->getStatus() < 600)) {
 			// render client error states with own layout => own #body-status id
+			$csp = new ContentSecurityPolicy();
 			$tmplparams = $response->getParams();
 			$tmplparams['bodyid'] = "body-status";
 			$response->setParams($tmplparams);
+			$response->setContentSecurityPolicy($csp);
```
Comment (tsdicloud) [inline lib/Listener/BeforeTemplateRenderedListener.php:87]: I doubt that an "empty" Content Security Policy has any effect if no CSP rules are added? Can you explain that?

File: lib/Listener/BeforeTemplateRenderedListener.php [hunk 3]
```diff
@@ -72,14 +73,18 @@ public function handle(Event $event): void {
 		// FIXME as long as photos app has photos as internal label and appname label
 		if ($response->getRenderAs() === TemplateResponse::RENDER_AS_USER ||
 			 $response->getRenderAs() === TemplateResponse::RENDER_AS_PUBLIC) {
+			$csp = new ContentSecurityPolicy();
 			$this->fixPhotosAppName();
+			$response->setContentSecurityPolicy($csp);
```
Comment (tsdicloud) [inline lib/Listener/BeforeTemplateRenderedListener.php:78]: I doubt that an "empty" Content Security Policy has any effect if no CSP rules are added? Can you explain that?

**Review by tsdicloud — CHANGES_REQUESTED** [review]
Bitte auch die Codeteile thematisch zusammenhalten:
```
			$tmplparams = $response->getParams();
			$tmplparams['bodyid'] = "body-status";
			$response->setParams($tmplparams);
			
			$csp = new ContentSecurityPolicy();
			$response->setContentSecurityPolicy($csp);
```

File: lib/Listener/BeforeTemplateRenderedListener.php (hunk 2 above)
Comment (liviu-preda-tsystems) [reply lib/Listener/BeforeTemplateRenderedListener.php]: I have refactored the code, moving all relevant CSP Header setters into an event listener for Content Security Policy changes (which should happen every time a file is directly downloaded). Not sure the BeforeTemplateRenderedListener would have caught file download events, as that does not trigger the re-rendering of the template. Thank you to check and let me know what you think.

**Review by tsdicloud — APPROVED** [review]
This looks very clean now.THX!
We must see whether this helps.

**Pull request merged**